geode-core/src/main/java/org/apache/geode/distributed/internal/DistributionAdvisor.java

```diff
@@ -172,6 +174,10 @@
    */
   private long currentVersionOpCount;
 
+  private Map<Thread, ExceptionWrapper> currentVersionOperationThreads = new HashMap<>();
```

Comment:
One other thought about how to refactor this code. It would be nice to not always have these HashMap instances on every advisor since they are only used of debug logging is enabled.
It would also be helpful if these things were all on a single class of their own:
  opCountLock, currentVersionOpCount, previousVersionOpCount, currentVersionOperationThreads, previousVersionOperationThreads

If you refactored all into their own class then that class could replace the opCountLock (since you could just naturally synchronize on "this" in that class. You could create a different version of this class when it is constructed if debug logging is enabled that has the new hashmaps and knows about the new ExceptionWrapper class

Reply:
I like this idea - it simplifies the code quite a bit.  I'm moving the membershipVersion to the new class, too, so we can remove some of the synchronization on DistributionAdvisor and use a sync on the new object.